source/Missions/Services/Agents/Messages/AgentMovement.cs

```diff
@@ -0,0 +1,185 @@
+﻿using Common.PacketHandlers;
+using LiteNetLib;
+using Missions.Services.Agents.Packets;
+using ProtoBuf;
+using SandBox.Missions.MissionLogics;
+using System;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Messages
+{
+    /// <summary>
+    /// Represents the delta of an <see cref="Agent"/>'s movement between the time it was created and the time it was last updated.
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public class AgentMovement : IPacket
```

Comment:
Convert this to a `public readonly struct` if possible

Reply:
I converted it to a struct. Can't make it readonly because of the `CalculateMovement` methods which do write to them.